stock_barcodes/models/stock_picking.py

```diff
@@ -46,7 +47,38 @@ def button_validate(self):
             and not self.move_line_ids.mapped("result_package_id")
         ):
             self.action_put_in_pack()
-        res = super().button_validate()
+        create_backorder = False
+        # Variable initialized as True to optimize break loop
+        skip_backorder = True
+        if self.env.context.get("stock_barcodes_validate_picking", False):
+            # Avoid backorder when all move lines are processed (done or done_forced)
+            prec = self.env["decimal.precision"].precision_get(
+                "Product Unit of Measure"
+            )
+            for move in self.move_lines.filtered(lambda sm: sm.state != "cancel"):
+                if (
+                    float_compare(
+                        move.quantity_done, move.product_uom_qty, precision_digits=prec
+                    )
+                    < 0
+                ):
+                    # In normal conditions backorder will be created
+                    create_backorder = True
+                    if not move.move_line_ids or any(
+                        sml.state in ["pending"] for sml in move.move_line_ids
+                    ):
+                        # If any move are not processed we can not skip backorder
+                        skip_backorder = False
+                        break
```

Comment:
Any line will impact in the whole picking?

Reply:
@sergio-teruel can you answer this? The problem I see is that this aborts the evaluation of the rest of the picking. Is that correct?

Reply:
ping @sergio-teruel